File: core/src/com/mygdx/game/characters/State/Stand.java
package com.mygdx.game.characters.State;

import com.mygdx.game.characters.GameCharacter;

public class Stand extends State {

    public Stand(GameCharacter character) {
        super(character);
    }

    public void stand() {}
    public void runLeft() {character.state = new RunningLeft(character);}
    public void runRight() {character.state =  new RunningRight(character);}
    public void walkLeft() {}
    public void walkRight() {}

    @Override
    public void climbUp() {character.state = new ClimbingUp(character);}

    @Override
    public void climbDown() {character.state = new ClimbingDown(character);}
}

File: web/src/characters/state/Stand.ts
import { State } from './State';
import { RunningLeft } from './RunningLeft';
import { RunningRight } from './RunningRight';
import { ClimbingUp } from './ClimbingUp';
import { ClimbingDown } from './ClimbingDown';

export class Stand extends State {
  stand(): void {}
  runLeft(): void {
    this.character.fsm = new RunningLeft(this.character);
  }
  runRight(): void {
    this.character.fsm = new RunningRight(this.character);
  }
  walkLeft(): void {}
  walkRight(): void {}
  climbUp(): void {
    this.character.fsm = new ClimbingUp(this.character);
  }
  climbDown(): void {
    this.character.fsm = new ClimbingDown(this.character);
  }
}
